dev-tools/omdb/src/bin/omdb/db/ereport.rs

```diff
@@ -464,123 +378,59 @@ async fn cmd_db_ereporters(
-        );
-        Vec::new()
-    } else {
-        (*conn).transaction_async({
-            let serial = serial.clone();
-            async move |conn| {
-                // Selecting all reporters may require a full table scan, depending
-                // on filters.
-                conn.batch_execute_async(ALLOW_FULL_TABLE_SCAN_SQL).await?;
-                let mut query = host_dsl::host_ereport
-                    .group_by((
-                        host_dsl::restart_id,
-                        host_dsl::sled_id,
-                        host_dsl::sled_serial,
-                    ))
-                    .select((
-                        host_dsl::restart_id,
-                        host_dsl::sled_id,
-                        host_dsl::sled_serial,
-                        min(host_dsl::time_collected),
-                        count_distinct(host_dsl::ena),
-                    ))
-                    .into_boxed();
-
-                if let Some(serial) = serial {
-                    query = query.filter(host_dsl::sled_serial.eq(serial.clone()));
-                }
-
-                query
-                    .load_async::<(Uuid, Uuid, String,  Option<DateTime<Utc>>, i64)>(
-                        &conn,
-                    )
-                    .await.context("listing host OS reporter entries")
-            }
-        }).await?
-    };
-
     #[derive(Tabled)]
     #[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
     struct ReporterRow {
         #[tabled(display_with = "datetime_opt_rfc3339_concise")]
         first_seen: Option<DateTime<Utc>>,
         id: Uuid,
-        identity: db::model::Reporter,
+        #[tabled(display_with = "display_option_error")]
+        identity: Option<Reporter>,
         #[tabled(display_with = "display_option_blank", rename = "S/N")]
         serial: Option<String>,
         #[tabled(display_with = "display_option_blank", rename = "P/N")]
         part_number: Option<String>,
         ereports: i64,
     }
 
-    let rows = {
-        let sp_rows = sp_ereporters.into_iter().map(
-            |(
-                restart_id,
-                slot,
-                sp_type,
-                serial,
-                part_number,
-                first_seen,
-                ereports,
-            )| {
-                ReporterRow {
-                    first_seen,
-                    identity: db::model::Reporter::Sp {
-                        slot: slot.0,
-                        sp_type: sp_type.into(),
-                    },
-                    serial,
-                    part_number,
-                    id: restart_id,
-                    ereports,
+    let mut rows = reporters
+        .into_iter()
+        .map(|(id, reporter, serial, part_number, first_seen, ereports)| {
+            let identity = match reporter.try_into() {
+                Ok(reporter) => Some(reporter),
+                Err(err) => {
+                    eprintln!("{err}");
```

Comment:
Here, and above on line 283 -- is just printing the error going to give enough context? Or do we want to provide some additional context?

(The `impl TryFrom<Reporter> for types::Reporter` messages seem pretty good, but far away from "what is omdb up to here, and what batch of data is it acting on")